src/rust/engine/task_executor/src/lib.rs

```diff
@@ -61,11 +61,11 @@ impl Executor {
   ///
   /// Dropping all clones of the Executor will cause the Runtime to shut down.
   ///
-  pub fn new_owned(core_threads: usize, num_threads: usize) -> Result<Executor, String> {
-    let runtime = Builder::new()
-      .core_threads(core_threads)
-      .max_threads(num_threads)
-      .threaded_scheduler()
+  // TODO(tokio-1.0): Figure out thread count configuration in Pants.
+  pub fn new_owned(_core_threads: usize, num_threads: usize) -> Result<Executor, String> {
+    let runtime = Builder::new_multi_thread()
+      .worker_threads(num_threads)
+      .max_blocking_threads(num_threads)
```

Comment:
The thread configuration for the multi-thread runtime changed in v1.0. Tokio no longer dynamically scales the number of threads as it does in prior versions. Instead, it will just spin up `worker_threads` threads and ensure that only `max_blocking_threads` are used for blocking work. (At least that is my naive take on the matter.)

The questions are:

1. What API should we be exposing for PyExecutor and related types?

2. The default number of threads now is the number of available cores. Should we just trust the Tokio defaults now and not perform manual configuration in the general case?

Reply:
According to the discussion on https://github.com/tokio-rs/tokio/issues/2802 this is now `worker_threads` + `max_blocking_threads`? Commented to confirm though: https://github.com/tokio-rs/tokio/issues/2802#issuecomment-763889643

Reply:
How should we express the concept of letting Tokio choose the `worker_threads` value while still retaining the ability to configure the number of blocking threads? Just do `max-core` and only use that value?

Reply:
> Tokio no longer dynamically scales the number of threads as it does in prior versions.

To be clear, the only thing that has changed here is how you set the upper limit on blocking threads. Just like in the previous versions, the number of core threads is fixed, and the number of blocking threads is dynamicly scaled until it reaches the limit.